### PR TITLE
switch none deployment strategy to basic

### PR DESCRIPTION
**Pull request by davidgamero**
ref #174

### CODE REVIEW

**Review by OliverMKing — CHANGES_REQUESTED** [review]

File: action.yml [hunk 1]
```diff
@@ -20,9 +20,9 @@ inputs:
       required: false
       default: true
    strategy:
-      description: 'Deployment strategy to be used. Allowed values are none, canary and blue-green'
+      description: 'Deployment strategy to be used. Allowed values are basic, canary and blue-green'
       required: false
```
[the remark this answers is not in the document]
Comment (OliverMKing) [reply action.yml:24]: There's also logic in the codebase that relies on the value of strategy with a switch statement. that should be modified away from fall through default logic.

**Review by jaiveerk — APPROVED** [review]

File: action.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidgamero) [reply action.yml]: swapped

**Comment by davidgamero** [comment]
changed it to only throw a warning for unrecognized strategies instead of running the basic one

File: README.md [hunk 2]
```diff
@@ -64,8 +64,8 @@ Following are the key capabilities of this action:
   </tr>
   <tr>
     <td>strategy </br></br>(Optional)</td>
-    <td>Acceptable values: none/canary/blue-green. <br>
-    Deployment strategy to be used while applying manifest files on the cluster.<br>none - No deployment strategy is used when deploying.<br>canary - Canary deployment strategy is used when deploying to the cluster.<br>blue-green - Blue-Green deployment strategy is used when deploying to cluster.</td>
+    <td>Acceptable values: basic/canary/blue-green. <br>
+    Default value: basic<br>Deployment strategy to be used while applying manifest files on the cluster.<br>basic - Template is force applied to all pods when deploying to cluster.<br>canary - Canary deployment strategy is used when deploying to the cluster.<br>blue-green - Blue-Green deployment strategy is used when deploying to cluster.</td>
```
Comment (OliverMKing) [inline README.md:68]: add that it's required here too

**Review by OliverMKing — CHANGES_REQUESTED** [review]

File: src/strategyHelpers/deploymentHelper.ts [hunk 3]
```diff
@@ -94,6 +91,10 @@ export async function deployManifests(
 
          return files
       }
+
+      default: {
+         core.warning('Deployment strategy is not recognized.')
```
[the remark this answers is not in the document]
Comment (davidgamero) [reply src/strategyHelpers/deploymentHelper.ts:96]: fixed

File: action.yml [hunk 4]
```diff
@@ -20,9 +20,9 @@ inputs:
...
       default: true
    strategy:
-      description: 'Deployment strategy to be used. Allowed values are none, canary and blue-green'
-      required: false
-      default: 'none'
+      description: 'Deployment strategy to be used. Allowed values are basic, canary and blue-green'
```
Comment (OliverMKing) [inline action.yml:23]: maybe specify that basic only works for deploy action

**Review by OliverMKing — APPROVED** [review]

**Pull request merged**